src/main/java/me/squidxtv/frameui/core/builder/ScreenBuilderImpl.java

```diff
@@ -0,0 +1,99 @@
+package me.squidxtv.frameui.core.builder;
+
+import me.squidxtv.frameui.api.ScreenRegistry;
+import me.squidxtv.frameui.api.builder.ScreenBuilder;
+import me.squidxtv.frameui.core.Renderer;
+import me.squidxtv.frameui.core.Screen;
+import me.squidxtv.frameui.core.ScreenLocation;
+import me.squidxtv.frameui.math.Direction;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.joml.Vector2i;
+
+import java.util.*;
+
+public class ScreenBuilderImpl implements ScreenBuilder {
+
+    private final Plugin plugin;
+    private final ScreenRegistry screenRegistry;
+    private final Vector2i size;
+    private final Set<Player> viewers;
+    private ScreenLocation screenLocation;
+    private Renderer renderer;
+    private String name;
+
+    public ScreenBuilderImpl(Plugin plugin, ScreenRegistry screenRegistry) {
+        this.plugin = plugin;
+        this.screenRegistry = screenRegistry;
+        this.size = new Vector2i(4, 4);
+        this.viewers = new HashSet<>();
+        this.name = "Screen";
+    }
+
+    @Override
+    public ScreenBuilder withWidth(int width) {
+        this.size.x = width;
+        return this;
+    }
+
+    @Override
+    public ScreenBuilder withHeight(int height) {
+        this.size.y = height;
+        return this;
+    }
+
+    @Override
+    public ScreenBuilder withSize(int width, int height) {
+        this.size.set(width, height);
+        return this;
+    }
+
+    @Override
+    public ScreenBuilder withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    @Override
+    public ScreenBuilder withViewer(Player player) {
+        this.viewers.add(player);
+
+        if (screenLocation != null) {
+            return this;
+        }
+        return withLocation(player.getLocation());
+    }
```

Comment:
Probably makes sense to add methods for `Collection<Player>` and varargs as well. You also should document such a behavior that if no location is set yet, it is going to use the player's location.

Reply:
Good point

Reply:
done